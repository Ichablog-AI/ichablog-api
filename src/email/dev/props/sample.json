{
    "name": "Angel",
    "url": "https://example.com/activate"
}
