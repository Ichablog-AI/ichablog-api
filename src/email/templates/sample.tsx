import { Body, Container, Head, Html, Link, Text } from '@react-email/components';
import type { EmailTemplate } from '../EmailTemplateInterface';

/**
 * Props for the sample email template.
 */
export type SampleEmailProps = {
    name: string;
    url: string;
};

/**
 * Generates the subject line for the sample email.
 *
 * @param props - The dynamic props for the email.
 * @returns Subject line as a string.
 */
const subject = ({ name }: SampleEmailProps) => `Welcome, ${name}!`;

/**
 * Generates the HTML body for the sample email.
 *
 * @param props - The dynamic props for the email.
 * @returns A JSX element representing the HTML email body.
 */
const html = ({ name, url }: SampleEmailProps) => (
    <Html>
        <Head />
        <Body>
            <Container>
                <Text>Hi {name},</Text>
                <Text>Click the link below to get started:</Text>
                <Link href={url}>Activate your account</Link>
            </Container>
        </Body>
    </Html>
);

/**
 * Generates the plain text version of the sample email.
 *
 * @param props - The dynamic props for the email.
 * @returns A plain text email string.
 */
const text = ({ name, url }: SampleEmailProps) =>
    `
Hi ${name},

Click the link below to get started:

${url}

Thanks,
Your App Team
`.trim();

/**
 * The sample email template combining subject, html, and text generators.
 */
export const SampleEmailTemplate: EmailTemplate<SampleEmailProps> = {
    subject,
    html,
    text,
};
