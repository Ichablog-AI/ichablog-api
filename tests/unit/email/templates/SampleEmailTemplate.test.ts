import { describe, expect, it } from 'bun:test';
import { SampleEmailTemplate } from '@be/email/templates/sample';
import { renderEmailTemplate } from '@be/test-helpers/renderEmailTemplate';

describe('SampleEmailTemplate', () => {
    it('should render correctly', async () => {
        const props = { name: 'Test User', url: 'https://example.com' };
        const { subject, html, text } = await renderEmailTemplate(SampleEmailTemplate, props);

        expect({ subject, html, text }).toMatchSnapshot();
    });
});
