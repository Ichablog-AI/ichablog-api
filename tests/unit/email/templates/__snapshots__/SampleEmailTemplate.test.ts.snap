// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`SampleEmailTemplate should match the expected snapshot 1`] = `
{
  "html": "<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><html dir="ltr" lang="en"><head><meta content="text/html; charset=UTF-8" http-equiv="Content-Type"/><meta name="x-apple-disable-message-reformatting"/></head><body><!--$--><table align="center" width="100%" border="0" cellPadding="0" cellSpacing="0" role="presentation" style="max-width:37.5em"><tbody><tr style="width:100%"><td><p style="font-size:14px;line-height:24px;margin-bottom:16px;margin-top:16px">Hi <!-- -->Snapshot Tester<!-- -->,</p><p style="font-size:14px;line-height:24px;margin-bottom:16px;margin-top:16px">Click the link below to get started:</p><a href="https://example.com/snapshot" style="color:#067df7;text-decoration-line:none" target="_blank">Activate your account</a></td></tr></tbody></table><!--7--><!--/$--></body></html>",
  "subject": "Welcome, Snapshot Tester!",
  "text": 
"Hi Snapshot Tester,

Click the link below to get started:

https://example.com/snapshot

Thanks,
Your App Team"
,
}
`;

exports[`SampleEmailTemplate should render correctly 1`] = `
{
  "html": "<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><html dir="ltr" lang="en"><head><meta content="text/html; charset=UTF-8" http-equiv="Content-Type"/><meta name="x-apple-disable-message-reformatting"/></head><body><!--$--><table align="center" width="100%" border="0" cellPadding="0" cellSpacing="0" role="presentation" style="max-width:37.5em"><tbody><tr style="width:100%"><td><p style="font-size:14px;line-height:24px;margin-bottom:16px;margin-top:16px">Hi <!-- -->Test User<!-- -->,</p><p style="font-size:14px;line-height:24px;margin-bottom:16px;margin-top:16px">Click the link below to get started:</p><a href="https://example.com" style="color:#067df7;text-decoration-line:none" target="_blank">Activate your account</a></td></tr></tbody></table><!--7--><!--/$--></body></html>",
  "subject": "Welcome, Test User!",
  "text": 
"Hi Test User,

Click the link below to get started:

https://example.com

Thanks,
Your App Team"
,
}
`;
